internal/session/repository_connection.go

```diff
@@ -266,31 +255,31 @@ func (r *ConnectionRepository) ConnectConnection(ctx context.Context, c ConnectW
 				// return err, which will result in a rollback of the update
 				return errors.New(ctx, errors.MultipleRecords, op, "more than 1 resource would have been updated")
 			}
-			newState, err := NewConnectionState(ctx, connection.PublicId, StatusConnected)
+			// Set the lower bound of the connected_time_range to indicate the connection is connected
+			rowsUpdated, err = w.Exec(ctx, connectConnection, []any{
+				sql.Named("public_id", c.ConnectionId),
+			})
 			if err != nil {
 				return errors.Wrap(ctx, err, op)
 			}
-			if err := w.Create(ctx, newState); err != nil {
-				return errors.Wrap(ctx, err, op)
-			}
-			connectionStates, err = fetchConnectionStates(ctx, reader, c.ConnectionId, db.WithOrder("start_time desc"))
-			if err != nil {
-				return errors.Wrap(ctx, err, op)
+			if rowsUpdated != 1 {
+				return errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("unable to connect connection %s", c.ConnectionId)))
 			}
 			return nil
 		},
 	)
 	if err != nil {
-		return nil, nil, errors.Wrap(ctx, err, op)
+		return nil, errors.Wrap(ctx, err, op)
 	}
-	return &connection, connectionStates, nil
+	connection.Status = StatusConnected.String()
```

Comment:
Can we read the status back from the database instead of just assuming this is correct if there were no errors?

Reply:
thanks!